pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -218,6 +218,26 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		glog.Errorf("NodeStageVolume: determineFilesystemType failed: %v", err)
 		return nil, err
 	}
+	// Check is something already mounted to stagingtargetPath
+	mounter := mount.New("")
+	mountedDev, _, err := mount.GetDeviceNameFromMount(mounter, stagingtargetPath)
+	if err != nil {
+		glog.Errorf("NodeStageVolume: Error getting device name for mount")
+		return nil, err
+	}
+	if mountedDev == device.Path {
+		glog.V(4).Infof("NodeStageVolume: Device %s already mounted to path %s", mountedDev, stagingtargetPath)
+		// Check are the capabilities compatible.
+		if existingFsType == requestedFsType {
```

Comment:
Can't we do this check just after determineFilesystemType() call?

Reply:
good point, these two checks can be combined. Also it seems, I have copied the code so that "err" that we return here, has no value. I combine those checks and force-push. 
This actually is place to review the functionality here.
The block of code after comment "what to do if existing file system is detected...",
I am not sure is in entirely right thing. If we detect same fsType file system, we do nothing,
that means a pod will get file system mounted with previous contents.
The current code is based on my early assumption "lets not destroy data". But giving a pod existing (probably old) contents also does not seem right.
Not sure what CSI spec says here, but I doubt does it determine what to do.
I was thinking, because we have shred on volume deletion, and we clear first 4k on creation, we should never get old data.
But I have seen in test logs that the line "Skip mkfs as %v file system already exists on %v" gets printed,
so there is possibility we get old data?